src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -324,10 +325,16 @@ def update_dashboard_filters(self, instance, validated_data):
         page_filter_keys = ["environment", "period", "start", "end", "utc"]
         dashboard_filter_keys = ["release"]
 
+        filters = {}
+
         if "projects" in validated_data:
-            instance.projects.set(validated_data["projects"])
+            if validated_data["projects"] == ALL_ACCESS_PROJECTS:
+                filters["all_projects"] = True
+            else:
+                if instance.filters and instance.filters.get("all_projects"):
+                    instance.filters["all_projects"] = False
```

Comment:
nit: just for consistency, should we avoid modifying instance.filter directly here and do
`filters["all_projects"] = False` and let filters be set on instances in line 347?

Reply:
Thanks for calling that out. It was why I moved `filters` up but I must've gone and tried something else at the time. 👍